Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -825,48 +880,44 @@ namespace GradientSignal
-                    return;
-                }
+            if (!assetId.IsValid())
+            {
+                // This case can occur either if the asset path is completely wrong, or if it's correct but the asset is still in
+                // the process of being created and being processed. Even though the second possibility is valid, 
+                AZ_Warning(
+                    "GradientSignal", false, "Can't find an Asset ID for %s, SetImageAssetPath() will be ignored.", assetPath.c_str());
+                return;
             }
+        }
 
-            AZ::Data::AssetBus::Handler::BusDisconnect(m_configuration.m_imageAsset.GetId());
+        // If we were given a valid asset, then make sure it is the right type
+        if (assetId.IsValid())
+        {
+            AZ::Data::AssetInfo assetInfo;
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequests::GetAssetInfoById, assetId);
 
+            if (assetInfo.m_assetType != azrtti_typeid<AZ::RPI::StreamingImageAsset>())
             {
-                // Only hold the lock during the actual data changes, to ensure that we aren't mid-query when changing it, but also to
-                // minimize the total lock duration. Also, because we've disconnected from the imageAsset Asset bus prior to locking this,
-                // we won't get any OnAsset* notifications while we're changing out the asset.
-                AZStd::unique_lock lock(m_queryMutex);
-
-                // Clear our cached image data
-                m_imageData = AZStd::span<const uint8_t>();
-
-                if (assetPath.empty())
-                {
-                    m_configuration.m_imageAsset.Reset();
-                }
-                else
-                {
-                    m_configuration.m_imageAsset = AZ::Data::AssetManager::Instance().FindOrCreateAsset(assetId, azrtti_typeid<AZ::RPI::StreamingImageAsset>(), m_configuration.m_imageAsset.GetAutoLoadBehavior());
-                }
+                AZ_Warning("GradientSignal", false, "Asset type for %s is not AZ::RPI::StreamingImageAsset, will be ignored", assetPath.c_str());
+                return;
             }
+        }
 
-            SetupDependencies();
-            m_configuration.m_imageAsset.QueueLoad();
-            AZ::Data::AssetBus::Handler::BusConnect(m_configuration.m_imageAsset.GetId());
-            LmbrCentral::DependencyNotificationBus::Event(GetEntityId(), &LmbrCentral::DependencyNotificationBus::Events::OnCompositionChanged);
+        AZ::Data::Asset<AZ::RPI::StreamingImageAsset> imageAsset;
+
+        if (assetId.IsValid())
+        {
+            imageAsset = AZ::Data::AssetManager::Instance().FindOrCreateAsset(
+                assetId, azrtti_typeid<AZ::RPI::StreamingImageAsset>(), m_configuration.m_imageAsset.GetAutoLoadBehavior());
         }
+
+        SetImageAsset(imageAsset);
```

Comment:
Previously you could pass an empty path to `SetImageAssetPath` to clear it since it would do `m_configuration.m_imageAsset.Reset();`. I think that should still work with this new logic since it will end up assigning `m_configuration.m_imageAsset` to an empty `AZ::Data::Asset<AZ::RPI::StreamingImageAsset>`, can you confirm? The `AutomatedTesting\Gem\PythonTests\largeworlds\gradient_signal\EditorScripts\ImageGradient_ModifiesSurfaces.py` automated test will fail otherwise since it exercises that usage.

Reply:
Verified that it still works both by running the automated test (which passed), and by manually calling SetImageAssetPath with an empty path in the Python Console so that I could watch it clear the asset from an Image Gradient.